packages/admin/admin/src/common/RowActions.tsx

```diff
@@ -0,0 +1,146 @@
+import { MoreVertical } from "@comet/admin-icons";
+import {
+    IconButton,
+    IconButtonProps,
+    ListItemIcon,
+    ListItemIconProps,
+    ListItemText,
+    ListItemTextProps,
+    Menu,
+    MenuItem,
+    MenuItemProps,
+    MenuProps,
+} from "@mui/material";
+import { createStyles, WithStyles, withStyles } from "@mui/styles";
+import * as React from "react";
+
+export interface RowActionsIconItemProps extends Omit<IconButtonProps, "children"> {
+    icon: React.ReactNode;
+}
+
+export interface RowActionsMenuItemComponentsProps {
+    listItemIcon?: Omit<ListItemIconProps, "children">;
+    listItemText?: Omit<ListItemTextProps, "primary">;
+}
+
+export interface RowActionsMenuItemProps extends Omit<MenuItemProps, "children" | "onClick"> {
+    icon?: React.ReactNode;
+    text: React.ReactNode;
+    onClick?: (e: React.MouseEvent<HTMLElement>, closeMenu: () => void) => void;
```

Comment:
It seems to me that you're calling `closeMenu` all the time. Why isn't this handled internally?

Reply:
There are some instances where the menu should not be closed immediately after clicking the item, e.g. when the item opens a sub-menu. But it probably makes sense to close by default and add a setting/prop to disable this for an individual item. I'll try to add something like this.

Reply:
@johnnyomair `closeMenu` is now called internally when clicking a `ListItem`.
This can be disabled and handled manually, as mentioned in the docs:

> ### Manually handle closing the menu
> 
> By default, the "More" menu is closed when clicking the item.  
> This can be prevented by setting `preventCloseOnClick` in the props object or by rendering a custom node.
> 
> The `closeMenu` function is passed as a parameter to the `onClick` function of the props object.  
> When rendering a custom node, `closeMenu` is available when passing a function that returns the node.